toolRepository-remote/src/test/scala/com/codacy/toolRespository/remote/ToolRepositoryRemoteSpec.scala

```diff
@@ -0,0 +1,256 @@
+import org.mockito.Mockito._
+import org.specs2.mock.Mockito
+import org.specs2.mutable.Specification
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class ToolRepositoryRemoteSpec extends Specification with Mockito {
+  implicit val actorSystem = ActorSystem("MyTest")
+  implicit val materializer = ActorMaterializer()
+
+  val toolA = Tool(
+    uuid = "uuid - A",
+    name = "name A",
+    shortName = "shortName",
+    documentationUrl = None,
+    sourceCodeUrl = None,
+    prefix = None,
+    needsCompilation = false,
+    configFilenames = Vector.empty,
+    version = "version",
+    description = None,
+    dockerImage = "dockerImage",
+    languages = Vector.empty,
+    clientSide = false,
+    enabled = true,
+    configurable = false)
+
+  val toolB = Tool(
+    uuid = "uuid - B",
+    name = "name B",
+    shortName = "shortName",
+    documentationUrl = None,
+    sourceCodeUrl = None,
+    prefix = None,
+    needsCompilation = false,
+    configFilenames = Vector.empty,
+    version = "version",
+    description = None,
+    dockerImage = "dockerImage",
+    languages = Vector.empty,
+    clientSide = false,
+    enabled = true,
+    configurable = false)
+
+  "list" should {
+    def eitherListToolsResponse(
+      listToolsResponse: ListToolsResponse): EitherT[Future, Either[Throwable, HttpResponse], ListToolsResponse] = {
+      val responseEither: Either[Either[Throwable, HttpResponse], ListToolsResponse] = Right(listToolsResponse)
+      EitherT(Future.successful(responseEither))
+    }
+
+    "return the list of tools" in {
+      val mockedClient = mock[ToolsClient]
+      val toolRepository = new ToolRepositoryRemote(mockedClient)
+
+      when(mockedClient.listTools(cursor = None)).thenReturn(
+        eitherListToolsResponse(ListToolsResponse.OK(ToolListResponse(None, Vector(toolA)))),
+        eitherListToolsResponse(ListToolsResponse.OK(ToolListResponse(None, Vector(toolB)))))
+
+      val tools = toolRepository.list()
+
+      // toolB should not be returned because the first one returned an empty cursor
+      tools must haveLength(1)
+      tools.head.uuid must_== (toolA.uuid)
+    }
+
+    "return list with multiple tools" in {
+      val mockedClient = mock[ToolsClient]
+      val toolRepository = new ToolRepositoryRemote(mockedClient)
+
+      val paginationInfo = PaginationInfo(Some("cursor"), Some(100), Some(1))
+
+      when(
+        mockedClient.listTools(
+          cursor = ArgumentMatchers.any[Option[String]],
+          limit = ArgumentMatchers.any[Option[Int]],
+          languages = ArgumentMatchers.any[Option[Iterable[String]]],
+          headers = ArgumentMatchers.any[List[HttpHeader]])).thenReturn(
+        eitherListToolsResponse(ListToolsResponse.OK(ToolListResponse(Some(paginationInfo), Vector(toolA)))),
+        eitherListToolsResponse(ListToolsResponse.OK(ToolListResponse(None, Vector(toolB)))))
+
+      val tools = toolRepository.list()
```

Comment:
If you remove the `Await` in the implementation would you make this using something comparable to `Async**Spec` in scalatest?

Reply:
I hope so. But specs2 was a pain until now. So I hope that switch is easier 😄

Reply:
We could consider using Scalatest for new tests since it is being adopted everywhere in Codacy.
What do you think @rtfpessoa ?